payments-core/src/main/java/com/stripe/android/paymentsheet/PaymentSheetViewModel.kt

```diff
@@ -288,8 +292,10 @@ internal class PaymentSheetViewModel @Inject internal constructor(
     }
 
     fun checkout(checkoutIdentifier: CheckoutIdentifier) {
-        // Clear out any previous errors before setting the new button to get updates.
-        _viewState.value = PaymentSheetViewState.Reset(null)
+        if (this.checkoutIdentifier != checkoutIdentifier) {
+            // Clear out any previous errors before setting the new button to get updates.
+            _viewState.value = PaymentSheetViewState.Reset()
```

Comment:
This comments says it clears previous errors, but it seems like ti doesn't because in the activities we are also clearing the error message?